sql/updates/2.6.0--2.5.2.sql

```diff
@@ -0,0 +1,119 @@
+
+-- Actually drop the table.
+-- ALTER EXTENSION only removes the table from the extension but doesn't drop it.
+DROP TABLE IF EXISTS _timescaledb_catalog.continuous_aggs_bucket_function;
+
+-- Drop overloaded versions of invalidation_process_hypertable_log() and invalidation_process_cagg_log()
+-- with bucket_functions argument.
+
+ALTER EXTENSION timescaledb DROP FUNCTION _timescaledb_internal.invalidation_process_hypertable_log(
+    mat_hypertable_id INTEGER,
+    raw_hypertable_id INTEGER,
+    dimtype REGTYPE,
+    mat_hypertable_ids INTEGER[],
+    bucket_widths BIGINT[],
+    max_bucket_widths BIGINT[],
+    bucket_functions TEXT[]
+);
+
+DROP FUNCTION IF EXISTS _timescaledb_internal.invalidation_process_hypertable_log(
+    mat_hypertable_id INTEGER,
+    raw_hypertable_id INTEGER,
+    dimtype REGTYPE,
+    mat_hypertable_ids INTEGER[],
+    bucket_widths BIGINT[],
+    max_bucket_widths BIGINT[],
+    bucket_functions TEXT[]
+);
+
+ALTER EXTENSION timescaledb DROP FUNCTION _timescaledb_internal.invalidation_process_cagg_log(
+    mat_hypertable_id INTEGER,
+    raw_hypertable_id INTEGER,
+    dimtype REGTYPE,
+    window_start BIGINT,
+    window_end BIGINT,
+    mat_hypertable_ids INTEGER[],
+    bucket_widths BIGINT[],
+    max_bucket_widths BIGINT[],
+    bucket_functions TEXT[],
+    OUT ret_window_start BIGINT,
+    OUT ret_window_end BIGINT
+);
+
+DROP FUNCTION IF EXISTS _timescaledb_internal.invalidation_process_cagg_log(
+    mat_hypertable_id INTEGER,
+    raw_hypertable_id INTEGER,
+    dimtype REGTYPE,
+    window_start BIGINT,
+    window_end BIGINT,
+    mat_hypertable_ids INTEGER[],
+    bucket_widths BIGINT[],
+    max_bucket_widths BIGINT[],
+    bucket_functions TEXT[],
+    OUT ret_window_start BIGINT,
+    OUT ret_window_end BIGINT
+);
+
+--undo compression feature for caggs
+--check that all continuous aggregates have compression disabled
+--this check is sufficient as we cannot have compressed chunks, if
+-- compression is disabled
+DO $$
+DECLARE
+  cagg_name NAME;
+  cnt INTEGER := 0;
+BEGIN
+    FOR cagg_name IN
+        SELECT view_name,
+               materialization_hypertable_schema,
+               materialization_hypertable_name
+        FROM timescaledb_information.continuous_aggregates
+        WHERE compression_enabled is TRUE
+    LOOP
+        RAISE NOTICE 'Compression is enabled for continuous aggregate: %', cagg_name;
+        cnt := cnt + 1;
+    END LOOP;
+    IF cnt > 0 THEN
+       RAISE EXCEPTION 'Cannot downgrade as compression is enabled for continuous aggregates' 
+            USING DETAIL = 'Please disable compression on all continuous aggregates before downgrading',
+            HINT = 'To disable compression, call decompress_chunk to decompress chunks, then drop any existing compression policy on the continuous aggregate, and finally run ALTER MATERIALIZED VIEW % SET timescaledb.compress = ''false'' ';
```

Comment:
Nit, make error messages conformant:

```suggestion
       RAISE EXCEPTION 'cannot downgrade as compression is enabled for continuous aggregates' 
            USING DETAIL = 'Please disable compression on all continuous aggregates before downgrading.',
            HINT = 'To disable compression, call decompress_chunk to decompress chunks, then drop any existing compression policy on the continuous aggregate, and finally run ALTER MATERIALIZED VIEW % SET timescaledb.compress = ''false''. ';
```